src/rascal/math/kvec_generator.hh

```diff
@@ -120,14 +120,14 @@ namespace rascal {
        * row(i) = i-th vector. If the origin k=(0,0,0) is included, it is
        * always stored in row(0).
        */
-      Matrix_t get_kvectors() const {
+      Eigen::Ref<const Matrix_t> get_kvectors() {
```

Comment:
the function can still be declared as const! Any specific reason why you removed the qualifier?